adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -27,6 +27,9 @@ type breakCleanUpTombstonedNodesFunc func() bool
 // Delete attaches a tombstone to an item so it can be periodically cleaned up
 // later and the edges reassigned
 func (h *hnsw) Delete(id uint64) error {
+	h.deleteVsInsertLock.Lock()
+	defer h.deleteVsInsertLock.Unlock()
```

Comment:
maybe add a comment about the lock strategy?

Reply:
Great idea!

Reply:
Explanation added [here](https://github.com/semi-technologies/weaviate/pull/2191/commits/d271455415b95a5f5e95398e6b3859e9d0fd8fc1).